py/server/deephaven/experimental/iceberg.py

```diff
@@ -233,13 +510,82 @@ def table(self, instructions: Optional[IcebergReadInstructions] = None) -> Icebe
         """
 
-        if instructions is not None:
+        if instructions:
             return IcebergTable(self.j_object.table(instructions.j_object))
         return IcebergTable(self.j_object.table())
 
+    def table_writer(self, writer_options: TableWriterOptions):
+        """
+        Create a new IcebergTableWriter for this Iceberg table using the provided TableWriterOptions.
+        This method will perform schema validation to ensure that the provided table definition from the writer options
+        is compatible with the Iceberg table schema. All further writes performed by the returned writer will not be
+        validated against the table's schema, and thus would be faster.
+
+        Creating a table writer is the recommended approach if users want to write to the same iceberg table multiple
+        times.
+
+        Args:
+            writer_options: The options to configure the table writer.
+
+        Returns:
+            the table writer object
+        """
+
+    def append(self, instructions: IcebergParquetWriteInstructions):
+        """
+        Append the provided Deephaven tables from the write instructions as new partitions to the existing Iceberg
+        table in a single snapshot. This will not change the schema of the existing table.
+
+        This method will create a new IcebergTableWriter with the provided table definition from the write instructions,
+        and use that writer to write the data to the table. Therefore, this method is not recommended if users want to
+        write to the iceberg table multiple times. Instead, users should create a single IcebergTableWriter and use it
+        to append multiple times.
+
+        Args:
+            instructions (IcebergParquetWriteInstructions): the instructions for customizations while writing.
+        """
+        self.j_object.append(instructions.j_object)
+
+    def overwrite(self, instructions: IcebergParquetWriteInstructions):
+        """
+        Overwrite the existing Iceberg table with the provided Deephaven tables from the write instructions in a
+        single snapshot. This will delete all existing data but will not change the schema of the existing table.
+        Overwriting a table while racing with other writers can lead to failure/undefined results.
+
+        This method will create a new IcebergTableWriter with the provided table definition from the write instructions,
+        and use that writer to write the data to the table. Therefore, this method is not recommended if users want to
+        write to the iceberg table multiple times. Instead, users should create a single IcebergTableWriter and use it
+        to write multiple times.
+
+        Args:
+            instructions (IcebergParquetWriteInstructions): the instructions for customizations while writing.
+        """
+        self.j_object.overwrite(instructions.j_object)
+
+    def write_data_file(self, instructions: IcebergParquetWriteInstructions) -> list:
```

Comment:
`list` is a generic type, we might want to provide a specific type for the actual return file.

Reply:
Discussed with @devinrsmith, we are removing the write_data_file API from python for now and can add back if required.